pkg/util/validation/errors.go

```diff
@@ -293,16 +293,20 @@ func NewMaxQueryExpressionSizeBytesError(actualSizeBytes, maxQuerySizeBytes int)
 		maxQueryExpressionSizeBytesFlag))
 }
 
-func NewRequestRateLimitedError(limit float64, burst int) LimitError {
-	return LimitError(globalerror.RequestRateLimited.MessageWithPerTenantLimitConfig(
+func FormatRequestRateLimitedError(limit float64, burst int) string {
+	return globalerror.RequestRateLimited.MessageWithPerTenantLimitConfig(
 		fmt.Sprintf("the request has been rejected because the tenant exceeded the request rate limit, set to %v requests/s across all distributors with a maximum allowed burst of %d", limit, burst),
-		requestRateFlag, requestBurstSizeFlag))
+		requestRateFlag,
+		requestBurstSizeFlag,
+	)
 }
 
-func NewIngestionRateLimitedError(limit float64, burst int) LimitError {
-	return LimitError(globalerror.IngestionRateLimited.MessageWithPerTenantLimitConfig(
+func FormatIngestionRateLimitedError(limit float64, burst int) string {
```

Comment:
Since you're touching this function names, I'd do: `FormatRequestRateLimitedErrorMessage` or simply `FormatRequestRateLimitedMessage`.
We're not returning an error here, but a message.